docs/389ds/howto/howto-test-haproxy.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "Testing HAProxy with 389 DS"
+---
+
+# Testing HAProxy with 389 DS
+-----------------
+
+{% include toc.md %}
+
+## Introduction
+
+A simple guide for HAProxy with LDAP configuration for testing purposes. If used in production, make sure to use valid certificates (as opposed to self-signed used in the guide).
+
```

Comment:
As we may add other examples in the future,
I would add another chapter level:
## Example supporting simple authentication over ldaps 
Then
### Step 1: Create Virtual Machines 
...

I do not this it possible to perform GSSAPI or EXTERNAL (i.e certificate based) authentication over HAProxy with the current code but if someday it is possible, we want to document that in this howto.
And we may add the config on plain ldap (I wonder how the certificate validation for StartTLS would work through HAProxy - it probably does not work with current code ...)

Reply:
@vashirov may fix me, but HAProxy doesn't provide any easy and secure way for GSSAPI/EXTERNAL configuration if I understand correctly. It may forward what it gets from the client, but that's it.
And for StartTLS - the same's true.

We can have a task for future investigation, but I think we should go with SSL and recommend this to the 389 DS administrators.

Reply:
> 'We can have a task for future investigation, but I think we should go with SSL and recommend this to the 389 DS administrators.'

Sorry if I was not clear. I fully agree with you, my point is only to:
 - structure the document in order to allow it to evolve if needed.
 - Precise clearly that this document describes the use of HAProxy over ldaps. 
   It is an important point and it is not explicitly mentioned in the current version of the document.

FYI: About GSSAPI, I also do not think that there are any solutions, for EXTERNAL/StartTLS, IMHO there may be partial solution if NSS provides the right callbacks (but as you said, it is another story and another ticket !)

Reply:
> * structure the document in order to allow it to evolve if needed.
> * Precise clearly that this document describes the use of HAProxy over ldaps.
>   It is an important point and it is not explicitly mentioned in the current version of the document.

Got it and fully agree! I'll restructure the doc accordingly...